joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/ErbTests.scala

```diff
@@ -212,4 +189,162 @@ class ErbTests extends RubyCode2CpgFixture {
       case xs => fail(s"Expected one call to fmtString, got [${xs.code.mkString(",")}]")
     }
   }
+
+  // Test for lowered do-block
+  "Do-block in <%= %> tags" should {
+    /* The code gets lowered to:
+     * ```
+     *   joern__buffer = ""
+     *   joern__buffer << form_with(url: some_url)
+     *   rails_lambda_0 = lambda do |form|
```

Comment:
both `joern__buffer` and `rails_lambda_0` should have names that are invalid ruby identifiers so they can't possibly clash with real names from the program (e.g. `<output_buffer>` and `<rails_lambda>0`)

Reply:
We can adjust this in the `RubyJsonToNodeCreator` then. It isn't possible to give these variables invalid identifiers during the `ruby_ast_gen` phase, otherwise we won't get an AST from `ruby_ast_gen`

Reply:
No, at that point we'd just be fooling ourselves about whether clashes with names in the analyzed program are possible or not.

If there are technical reasons that we must parse these names as Ruby identifiers, then we just have to live with the potential name clashes as a limitation.